File: routes/GET/getAnimesOfAnStudio.py
from classes import animes
from flask import make_response, jsonify

def get(studio):
    try:
        anime = animes.Anime(studio=studio)
        getAnimes =anime.getAnimesOfAnStudio() 
        match getAnimes["message"]:
            case "No animes were found":
                resp = make_response(jsonify(getAnimes))
                resp.status_code = 404
                return resp
            case "Animes found":
                resp = make_response(jsonify(getAnimes))
                resp.status_code = 200
                return resp
            case "An error has occurred while getting the animes":
                resp = make_response(jsonify(getAnimes))
                resp.status_code = 200
                return resp
    
    except Exception as e:
        resp = make_response(jsonify({"message":"An error has occurred", "error":e.args}))
        resp.status_code = 500
        return resp

File: src/routes/GET/getAnimesOfAnStudio.ts
import { Anime } from "../../classes/animes";
import type { Response, Request } from "express";

export default function getAnimesOfAnStudio(req: Request, res: Response) {
  try {
    const studio = req.params.studio;
    const animes = new Anime(0, "", "", "", "", studio).getAnimesOfAnStudio();
    switch (animes.message) {
      case "success":
        res.statusCode = 200;
        res.json(animes);
        break;
      case "An error has occurred while getting the animes":
        res.statusCode = 500;
        res.json(animes);
    }
  } catch (error: any) {
    res.statusCode = 500;
    res.json({ message: "There was an error", error: error.message });
  }
}
